packages/editor/src/components/post-featured-image/with-panel-check.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { store as editorStore } from '../../store';
+import PostFeaturedImage from './index';
+import PostFeaturedImageCheck from './check';
+
+const PANEL_NAME = 'featured-image';
+
+function FeaturedImage() {
+	const isEnabled = useSelect(
+		( select ) => select( editorStore ).isEditorPanelEnabled( PANEL_NAME ),
+		[]
+	);
+	if ( ! isEnabled ) {
+		return null;
+	}
+	return (
+		<PostFeaturedImageCheck>
```

Comment:
This seems to be already used in the `PostFeaturedImage` component, so the only thing this wrapper adds is the isEnabled check, maybe we can bundle it instead of having a new component. What impact would that have?

Reply:
Maybe it's fine to keep the component but just reuse the existing `PostFeaturedImagePanel` (because it's still the same panel, we just moved it)

Reply:
I had considered adding a new prop, but for some reason I was stuck that the prop should be `as` - to render the wrapper. That way the default would be the panelBody and here that we don't want a toggle header we should pass a fragment or a div.. The problem I was seeing would be the name wouldn't be accurate anymore and is already public API. But after your nudge it came to me that we want is always a panel and the prop should be whether to render the wrapper or not. 

I'll update.